go-controller/pkg/config/config.go

```diff
@@ -154,16 +154,19 @@ var (
 
 	// MasterHA holds master HA related config options.
 	MasterHA = HAConfig{
-		ElectionLeaseDuration: 60,
-		ElectionRenewDeadline: 30,
-		ElectionRetryPeriod:   20,
+		ElectionRetryPeriod: 20,
+		// account for a 60 second server-side timeout + ElectionRetryPeriod = 80 seconds one try may take.
+		// add extra 10 sec for a potentially successful retry.
+		// this also covers 3 retries in case of a 10 seconds etcd timeout in case it is overloaded.
+		ElectionRenewDeadline: 90,
+		ElectionLeaseDuration: 100,
```

Comment:
wont this also result in it taking longer for a standby node to takeover if the primary node has failed?

Reply:
usually a leader should release the lease before exiting https://github.com/kubernetes/client-go/blob/master/tools/leaderelection/leaderelection.go#L303 and that should trigger re-election
but sometimes it may fail if apiserver or etcd is unresponsive (so if [Update](https://github.com/kubernetes/client-go/blob/master/tools/leaderelection/leaderelection.go#L307) request fails), but in that case a new leader probably won't be able to get or update the lease too anyway.
I also figured that we don't handle panics in `runOvnKube`, so I added that to make sure the lease is released before exiting

Reply:
but if the lease wasn't released, like power off of the node, or network outage. It would increase time it takes for standby node to take over?

@jcaamano can you ptal at this too. Good to get your thoughts on it.

Reply:
yep, and if we find that critical, I can propose
ElectionRenewDeadline=65 (that is still 3 etcd timeout retries and 0 server-side timeouts)
ElectionLeaseDuration=70 (that is 10 sec more compared to now)

worst-case we could go
ElectionRenewDeadline=55 (that is 2 etcd timeout retries compared to 1 right now
ElectionLeaseDuration=60

Reply:
With the current configuration:
```
LeaseDuration=60s
RenewDeadline=30s
RetryPeriod=20s
```
- a non leader would attempt to gain leadership after 60 seconds of the last observed leadership renewal, every 20 seconds
- a leader would try to renew leadership lease every 20 seconds and give up after 30 seconds. 

So yeah, with the current configuration, a leader would actually give up the first time it can't renew the lease without retries and that does not make a lot of sense.

I am not sure releasing the lease is relevant at all. Per the documentation, the mechanism is based on the last time the lease was renewed, not on the release (but maybe it is relevant although not documented):
```
	// A client needs to wait a full LeaseDuration without observing a change to
	// the record before it can attempt to take over. When all clients are
	// shutdown and a new set of clients are started with different names against
	// the same leader record, they must wait the full LeaseDuration before
	// attempting to acquire the lease. Thus LeaseDuration should be as short as
	// possible (within your tolerance for clock skew rate) to avoid a possible
	// long waits in the scenario.
```

There are recommended default values and reasoning to handle kube-apiserver disruption here:
https://github.com/openshift/enhancements/blob/84e894ead7b188a1013556e0ba6973b8463995f1/CONVENTIONS.md#handling-kube-apiserver-disruption
```
LeaseDuration=137s
RenewDealine=107s
RetryPeriod=26s
```

OVN-K SNO is already using these suggested values.

They seem high, but I would probably stick to those values unless we have a good reason not to, which we should doc somewhere and keep them consistent.

If we want to keep them lower, I would follow a parallel reasoning to the one that came up with those values. I think the values @npinaeva is suggesting do that (however fortuitous or not), proposing lower margins than the suggested values but keeping to the minimums established by that reasoning. In particular, I don't where the minimum tolerance of 60s for api server downtime comes from, but that would be tolerated with the proposed values.

Reply:
oh wow, I was completely unaware of that discussion, thanks a lot for finding that @jcaamano !
their reasoning is very similar to what I was thinking, but a bit more bold. So I +1 to follow openshift standard values.
p.s. @jcaamano releasing the lease sets `LeaseDuration` to 1s https://github.com/kubernetes/client-go/blob/master/tools/leaderelection/leaderelection.go#L303, that is why released lock causes quick re-election

Reply:
> I see. So that is for the client that holds the lease, the leader. The other question is: does this make non-leaders acquire the lease quicker than if we didn't release it?

I guess it does because that `LeaseDuration` update affects every client immediately?

Reply:
other potential leaders will try to acquire the lease on their next iteration, that happens every `RetryPeriod`.
```
	// A client needs to wait a full LeaseDuration without observing a change to
	// the record before it can attempt to take over.
```
so releasing the lock basically enables potential leader to acquire the lease, while otherwise they would need to wait for a full LeaseDuration seconds to expire.